### PR TITLE
Refactor tests so that it's easy to add v4 packages

### PR DESCRIPTION
**Pull request by larioj**
Refactor tests to use Table testing style. With this style we'll be able add new examples (v4 examples) to the tables instead of copying and pasting the tests.

Also moves TestingPackages to cosmos-model, so that all tests in all modules are able to use the example packages.

### CODE REVIEW

File: cosmos-bijection/src/test/scala/com/mesosphere/universe/bijection/UniverseConversionsSpec.scala [hunk 1]
```diff
@@ -138,8 +142,30 @@ final class UniverseConversionsSpec extends FreeSpec with Matchers {
     assertResult(v3)(v2FromV3.as[universe.v3.model.Version])
   }
 
-  private[this] def expectV3(v2model: universe.v2.model.PackageDetails): universe.v2.model.PackageDetails = v2model.copy(
-    packagingVersion = universe.v2.model.PackagingVersion("3.0")
+  def id(
```
Comment (cruhland) [inline cosmos-bijection/src/test/scala/com/mesosphere/universe/bijection/UniverseConversionsSpec.scala]: Could you give this a more descriptive name? Something like `metadataRoundTrip`? Yes, it has the same external behavior as the identity function, but the code is more complex than `x => x`.

File: cosmos-bijection/src/test/scala/com/mesosphere/universe/bijection/UniverseConversionsSpec.scala [hunk 2]
```diff
@@ -138,8 +142,30 @@ final class UniverseConversionsSpec extends FreeSpec with Matchers {
...
+    metadataAndReleaseVersion: (universe.v3.model.Metadata, universe.v3.model.ReleaseVersion)
+  ): (universe.v3.model.Metadata, universe.v3.model.ReleaseVersion) = {
+    metadataAndReleaseVersion
+      .as[universe.v3.model.SupportedPackageDefinition]
+      .as[(universe.v3.model.Metadata, universe.v3.model.ReleaseVersion)]
+  }
+
+  def almostId(
```
Comment (cruhland) [inline cosmos-bijection/src/test/scala/com/mesosphere/universe/bijection/UniverseConversionsSpec.scala]: More descriptive name please?

File: cosmos-model/src/test/scala/com/mesosphere/universe/v3/model/PackagingVersionTestOps.scala [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.mesosphere.universe.v3.model
+
+import com.mesosphere.universe
+import org.scalatest.prop.TableFor2
+
+object PackagingVersionTestOps {
```
Comment (jsancio) [inline cosmos-model/src/test/scala/com/mesosphere/universe/v3/model/PackagingVersionTestOps.scala]: FYI. The name of this object doesn't match the content, right?

Comment (larioj) [inline cosmos-model/src/test/scala/com/mesosphere/universe/v3/model/PackagingVersionTestOps.scala]: @jsancio What do you mean? It's test operations associated with Packaging Version.

Comment (cruhland) [inline cosmos-model/src/test/scala/com/mesosphere/universe/v3/model/PackagingVersionTestOps.scala]: Oh yeah, I forgot that I wanted to comment on this too. It's kind of a pattern in Scala that classes ending with `Ops` are for extension methods only.

Comment (cruhland) [inline cosmos-model/src/test/scala/com/mesosphere/universe/v3/model/PackagingVersionTestOps.scala]: For example: http://www.scala-lang.org/api/current/scala/collection/immutable/StringOps.html

Comment (larioj) [inline cosmos-model/src/test/scala/com/mesosphere/universe/v3/model/PackagingVersionTestOps.scala]: I'm going to rename this to TestCompanion.

**Pull request merged**